build_from_config

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env bash
+
+
+OPT=""
+VAL=""
+get_opt_and_val() {
+    IFS=":" read -ra TEMP <<< "${LINE}"
+    OPT="${TEMP[0]}"
+    VAL="${TEMP[1]}"
+}
+
+
+MIN=""
+MAX=""
+PLACEHOLDER=""
+COMMANDS=""
+get_min_and_max() {
+    IFS="," read -ra TEMP <<< "${VAL}"
+    MIN="${TEMP[0]/ /}"
+    MAX="${TEMP[1]/ /}"
+    if [ "${MIN}" != "" ]; then
+        COMMANDS="${COMMANDS} && [ ${PLACEHOLDER} -ge ${MIN} ]"
+    fi
+    if [ "${MAX}" != "" ]; then
+        COMMANDS="${COMMANDS} && [ ${PLACEHOLDER} -le ${MAX} ]"
+    fi
+}
+
+
+get_min_and_max_float() {
+    IFS="," read -ra TEMP <<< "${VAL}"
+    MIN="${TEMP[0]/ /}"
+    MAX="${TEMP[1]/ /}"
+    if [ "${MIN}" != "" ]; then
+        COMMANDS="${COMMANDS} && [ \$(echo \"${PLACEHOLDER} >= ${MIN}\" | bc) -eq 1 ]"
+    fi
+    if [ "${MAX}" != "" ]; then
+        COMMANDS="${COMMANDS} && [ \$(echo \"${PLACEHOLDER} <= ${MAX}\" | bc) -eq 1 ]"
+    fi
+}
+
+
+process_rbtype_command() {
+    case ${VAL} in
+        Real)
+            COMMANDS="${COMMANDS} && [[ @real == \"true\" ]] && [[ @binary == \"false\" ]]"
+            ;;
+        Complex)
+            COMMANDS="${COMMANDS} && [[ @real == \"false\" ]]"
+            ;;
+        Bool)
+            COMMANDS="${COMMANDS} && [[ @binary == \"true\" ]]"
+            ;;
+        Any)
+            ;;
+        *) 
+            echo "Not support ${VAL} in rbtype"
+            ;;
+    esac
+}
+
+
+process_structure_command() {
+    case ${VAL} in
+        Square)
+            COMMANDS="${COMMANDS} && [[ @rows == @cols ]]"
+            ;;
+        Rectangle)
+            COMMANDS="${COMMANDS} && [[ @rows != @cols ]]"
+            ;;
+        Symmetric)
+            COMMANDS="${COMMANDS} && [[ @real == true ]] && [[ @rows == @cols ]] && [[ @nsym == 1 ]]"
+            ;;
+        Unsymmetric)
+            COMMANDS="${COMMANDS} && [[ @rows == @cols ]] && [[ @nsym != 1 ]]"
+            ;;
+        Hermitian)
+            COMMANDS="${COMMANDS} && [[ @real == false ]] && [[ @rows == @cols ]] && [[ @nsym == 1 ]]"
+            ;;
+        Any)
+            ;;
+        *)
+            echo "Not support ${VAL} in structure"
+            ;;
+    esac
+}
+
+
+while IFS='' read -r LINE || [[ -n "$LINE" ]]; do
+    if [[ ! "$LINE" =~ ^\/\/.*$ ]]; then
```

Comment:
Do you actually need the dot `.` here? You are trying to ignore `//` at the beginning of a line, but what if it was just `//` without any content? Then it would still be processed as far as I can see. I would just remove the dot, then it shuld be fine (the `$` is also not needed, but it does not hurt here):
```suggestion
    if [[ ! "$LINE" =~ ^\/\/*$ ]]; then
```

Reply:
I think it still needs `^\/\/.*`. This can also ignores `//` without any content.
If use `^//*$`, it only ignores `//`, `///` ... not `// comment`.

Reply:
You are right, according to [this documentation I found](https://www.tldp.org/LDP/abs/html/x17129.html), `*` behaves exactly as you said. I thought it would be similar to [globbing](https://www.tldp.org/LDP/abs/html/globbingref.html) inside the `if [[ ]]`, by bad.